### ISSUE TITLE
bug: lib doesn't work from node_modules - cannot read property 'state' of undefined

### ISSUE THREAD

**Issue by RikuVan** [tkh44/preact-smitty#2]
First, thanks for Smitty. I have really enjoyed using it--simple, small and easy to pick up. I have had to copy Preact-smitty directly into my project, however, to use it. If I use the dist version from node_modules, I get the following error: 

<img width="999" alt="screen shot 2018-01-07 at 11 02 52" src="https://user-images.githubusercontent.com/7317771/34648157-62fbd376-f39d-11e7-8a72-ac0f264aea28.png">

A minimal version is reproduced here: [https://glitch.com/edit/#!/handsomely-twister](https://glitch.com/edit/#!/handsomely-twister)

I am using it with the Preact-cli here: [https://github.com/RikuVan/webdevandsausages.v2](https://github.com/RikuVan/webdevandsausages.v2)

**Comment by RikuVan** [comment]
ok now the minimal version actually works fine, so it doesn't help explain why it doesn't work in my project. Maybe something about how the Preact-cli builds stuff?

**Comment by tkh44** [comment]
There are several bugs in this implementation that should have been fixed a long time ago. I didn't know anyone was still using this :)

If you were to port over the React version (https://github.com/tkh44/react-smitty/blob/master/src/index.js) to Preact you would probably have a much better time.

I wrote the React version after this one and fixed several problems that I never came back and applied to preact-smitty. If you would like to put in a PR to update this library I would happily merge it.
